calico_node/startup/startup.go

```diff
@@ -314,11 +333,16 @@ func configureIPsAndSubnets(node *api.Node) {
 		}
 	} else {
 		if ipv6Env != "" {
-			node.Spec.BGP.IPv6Address = parseIPEnvironment("IP6", ipv6Env, 6)
+			envIp6 := parseIPEnvironment("IP6", ipv6Env, 6)
+			if node.Spec.BGP.IPv6Address == nil || !node.Spec.BGP.IPv6Address.IP.Equal(envIp6.IP) {
+				checkConflicts = true
+			}
+			node.Spec.BGP.IPv6Address = envIp6
 		}
 		validateIP(node.Spec.BGP.IPv6Address)
 	}
 
```

Comment:
Would it be nicer and less code to store off the original value at the top of the function, and compare it to the new value at the bottom?

e.g.

```
oldv4 := node.Spec.BGP.IPv4Address
oldv6 := ...

// Rest of function.
...

if node.Spec.BGP.IPv4Addres != oldv4 ||  node.Spec.BGP.IPv6Address != oldv6 {
  ...
}
```